app/src/test/java/at/aau/edu/wizards/TooManyPlayerUnitTest.kt

```diff
@@ -0,0 +1,66 @@
+package at.aau.edu.wizards
+
+import at.aau.edu.wizards.api.Server
+import at.aau.edu.wizards.api.model.ServerConnection
+import at.aau.edu.wizards.ui.lobby.LobbyItem
+import at.aau.edu.wizards.ui.lobby.LobbyItemFactory
+import at.aau.edu.wizards.ui.lobby.LobbyViewModel
+import junit.framework.TestCase.assertFalse
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Assertions.assertTrue
+import org.junit.jupiter.api.Test
+import org.mockito.kotlin.mock
+
+class TooManyPlayerUnitTest {
+
+    private val server = mock<Server>()
+    private val lobbyItemFactory = mock<LobbyItemFactory>()
+    private val viewModel = LobbyViewModel(server, lobbyItemFactory)
+
+    @Test
+    fun `Testing numPlayer adding human CPU player`() {
+        viewModel.clicked(LobbyItem.AddCpu)
+        assertEquals(2, viewModel.numPlayer)
+    }
+
+    @Test
+    fun `Testing numPlayer adding human player`() {
+        val requestedConnection = mock<ServerConnection.ClientRequest>()
+        viewModel.clicked(LobbyItem.Requested(requestedConnection))
+        assertEquals(2, viewModel.numPlayer)
+
+    }
+
+    @Test
+    fun `Testing numPlayer adding human player and CUP player`() {
+        val requestedConnection = mock<ServerConnection.ClientRequest>()
+
+        viewModel.clicked(LobbyItem.Requested(requestedConnection))
+        viewModel.clicked(LobbyItem.AddCpu)
+        viewModel.clicked(LobbyItem.AddCpu)
+        viewModel.clicked(LobbyItem.Requested(requestedConnection))
+        viewModel.clicked(LobbyItem.AddCpu)
+        viewModel.clicked(LobbyItem.AddCpu)
+
+        assertEquals(6, viewModel.numPlayer)
+
+    }
+
+    @Test
+    fun `Testing numPlayer over 6 player`() {
+        val requestedConnection = mock<ServerConnection.ClientRequest>()
+
+        assertFalse(viewModel.getCheckTooManyPlayer())
+
+        viewModel.clicked(LobbyItem.Requested(requestedConnection))
+        viewModel.clicked(LobbyItem.AddCpu)
+        viewModel.clicked(LobbyItem.AddCpu)
+        viewModel.clicked(LobbyItem.Requested(requestedConnection))
+        viewModel.clicked(LobbyItem.AddCpu)
+        viewModel.clicked(LobbyItem.AddCpu)
+        viewModel.clicked(LobbyItem.AddCpu)
+
+        assertTrue(viewModel.getCheckTooManyPlayer())
+    }
+
```

Comment:
You missed the case where it's neither `AddCpu` or `Requested`. That's why we have reports showing it to us, please add that too. It's not a perfect test case right now, but good work 👍. Please add the missing piece and we achieve the goal of "having code coverage". 

<img width="581" alt="Screenshot 2023-05-12 at 21 30 48" src="https://github.com/lechld/Wizard/assets/68105512/0afe0378-ca87-40d3-9682-90ee44c10d6c">

Reply:
everything is now tested